### PR TITLE
feat: New Module `avm-res-compute-sshpublickey`

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
## Description

New module for `compute\sshPublicKey` - This is outside of the inital AVM release, but uses a script within the the `e2e-template-assets` folder so produced this module for testing.

| Status Badge |
|----------------|
| [avm.res.compute.ssh-public-key](https://github.com/ChrisSidebotham/bicep-registry-modules/actions/workflows/avm.res.compute.ssh-public-key.yml)|

### CODE REVIEW

File: avm/res/compute/ssh-public-key/README.md [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+| `location` | string | The location the resource was deployed into. |
+| `name` | string | The name of the Public SSH Key. |
+| `resourceGroupName` | string | The name of the Resource Group the Public SSH Key was created in. |
+| `resourceId` | string | The resource ID of the Public SSH Key. |
+
+## Cross-referenced modules
+
+_None_
+
+## Deployment examples
```
Comment (AlexanderSehr) [inline avm/res/compute/ssh-public-key/README.md:184]: Please regenerate the docs (i.e.. delete and generate). This is half CARML half AVM because the utility tries to preserve the original headers

File: avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.sshPublicKeys-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'cspkdef'
```
Comment (AlexanderSehr) [inline avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep:15]: Need to figure this out 🍡

Comment (AlexanderSehr) [reply avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep:15]: Alright, it's min

Comment (AlexanderSehr) [reply avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep:15]:
```suggestion
param serviceShort string = 'cspmin'
```

**Review by eriqua — CHANGES_REQUESTED** [review]
Please link successful pipeline badge

**Comment by ChrisSidebotham** [comment]
> Please link successful pipeline badge

Do we have a template to be published in this repo for this in development yet?

File: avm/res/compute/ssh-public-key/README.md (hunk 1 above)
Comment (ChrisSidebotham) [reply avm/res/compute/ssh-public-key/README.md]: Re-generated docs

File: avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep (hunk 2 above)
Comment (ChrisSidebotham) [reply avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep]: updated in lastest commit

File: .vscode/settings.json [hunk 3]
```diff
@@ -12,5 +12,6 @@
   "[markdown]": {
     "editor.defaultFormatter": "esbenp.prettier-vscode"
   },
-  "editor.formatOnSave": true
+  "editor.formatOnSave": true,
+  "sarif-viewer.connectToGithubCodeScanning": "off"
```
Comment (eriqua) [inline .vscode/settings.json:16]: Is this change intended? This will impact the whole repo, not avm specific

File: avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,45 @@
+targetScope = 'subscription'
+
```
Comment (eriqua) [inline avm/res/compute/ssh-public-key/tests/e2e/defaults/main.test.bicep:2]:
```suggestion

metadata name = 'Using only defaults'
metadata description = '''
This instance deploys the module with the minimum set of required parameters.
> **Note:** The test currently implements additional non-required parameters to cater for a test-specific limitation.
'''
```

File: avm/res/compute/ssh-public-key/tests/e2e/max/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,71 @@
+targetScope = 'subscription'
+
```
Comment (eriqua) [inline avm/res/compute/ssh-public-key/tests/e2e/max/main.test.bicep:2]:
```suggestion

metadata name = 'Using large parameter set'
metadata description = 'This instance deploys the module with most of its features enabled.'
```

File: avm/res/compute/ssh-public-key/tests/e2e/waf-aligned/main.test.bicep [hunk 6]
```diff
@@ -0,0 +1,72 @@
+targetScope = 'subscription'
+
```
Comment (eriqua) [inline avm/res/compute/ssh-public-key/tests/e2e/waf-aligned/main.test.bicep:2]:
```suggestion

metadata name = 'WAF-aligned'
metadata description = 'This instance deploys the module in alignment with the best-pratices of the Azure Well-Architected Framework.'

```

File: avm/res/compute/ssh-public-key/README.md
Comment (eriqua) [inline avm/res/compute/ssh-public-key/README.md:1]: Please regen after applying above changes

**Review by ChrisSidebotham — COMMENTED** [review]
🆗

**Review by AlexanderSehr — APPROVED** [review]

**Review by eriqua — APPROVED** [review]
